src/utils/framework.ts

```diff
@@ -399,3 +416,69 @@ export const getMatrix2dToc = (framework: FrameworkFields | undefined): MatrixTo
 
     return toc.filter(isDefined);
 };
+
+export function getScaleWidgetsData(framework: FrameworkFields, entry: Entry) {
+    if (!framework || !framework.widgets) {
+        return [];
+    }
+
+    const { widgets } = framework;
+
+    const scaleWidgets = widgets
+        .filter(w => w.widgetId === 'scaleWidget')
+        .map((w) => {
+            const attributeData = entry.attributes[w.id];
+            const { properties: { data: { scaleUnits } } } = w as WidgetElement<ScaleWidget>;
+            if (isWidgetData(attributeData)) {
+                const value = scaleUnits.find(v => v.key === attributeData.data.value);
+                return value;
+            }
+            return undefined;
+        })
+        .filter(isDefined);
+
+    const scaleWidgetsInsideConditionals = widgets
+        .filter(w => w.widgetId === 'conditionalWidget')
+        .map((conditional) => {
+            const {
+                id,
+                properties: {
+                    data: {
+                        widgets: widgetsInsideConditional = [],
+                    } = {},
+                } = {},
+            } = conditional as WidgetElement<ConditionalWidget>;
+
+            return widgetsInsideConditional
+                .filter(w => w.widget && w.widget.widgetId === 'scaleWidget')
+                .map(({ widget }) => {
+                    const widgetAttributeData = entry
+                        .attributes[id]?.data?.value as ConditionalAttribute;
+
+                    if (widgetAttributeData?.selectedWidgetKey) {
+                        const { selectedWidgetKey } = widgetAttributeData;
+
+                        if (selectedWidgetKey) {
+                            const attributeData = widgetAttributeData[selectedWidgetKey];
+
+                            if (attributeData && isWidgetData(attributeData)) {
+                                const {
+                                    properties: {
+                                        data: {
+                                            scaleUnits,
+                                        },
+                                    },
+                                } = widget as WidgetElement<ScaleWidget>;
+
+                                return scaleUnits.find(v => v.key === attributeData.data.value);
+                            }
+                        }
+                    }
+                    return undefined;
```

Comment:
Return undefined if selectedWidgetKey is not defined immediately, will reduce a level of nesting